Improve manifest information (#93)
SAP-samples/ui5-typescript-walkthrough: steps/10/README.md

## Step 10: Manifest (Descriptor for Applications)

> All application-specific configuration settings will now further be put into the manifest. This clearly separates the application coding from the configuration settings and makes our app even more flexible.

Reviewer: ```suggestion
All application-specific configuration settings will now be put into the manifest. This clearly separates the application coding from the configuration settings and makes our app even more flexible.
```

-   `description`: Similarly, we make the description text language-dependent by referencing the `appDescription` text from the resource bundle using the handlebar syntax: {{key}}

> :warning: **Remember:** <br>

> > Properties of the resource bundle are enclosed in two curly brackets in the manifest. This is not an OpenUI5 data binding syntax, but a variable reference to the resource bundle in the manifest in handlebars syntax.

Reviewer: ```suggestion
> Properties of the resource bundle are enclosed in two curly brackets in the manifest. This is not an OpenUI5 data binding syntax but a variable reference to the resource bundle in the manifest in handlebars syntax.
```

In addition to the `sap.app` namespace, there are two other important namespaces:
The **`sap.ui`** namespace is used for UI-specific attributes and comes with the following main attributes:

> -   `technology` \(mandatory\): This property specifies the technology used for the application; the value is `UI5`.

Reviewer: ```suggestion
-   `technology` \(mandatory\): This property specifies the technology used for the application; its value must be `UI5`.
```

    -   The `minUI5Version` property is mandatory as it specifies the minimum version of OpenUI5 that the component requires to function properly. Following the guideline discussed in step 1, our component should be designed to work with the long-term support (LTS) version 1.120 of OpenUI5.
    
    -    The `libs` settings is used to declare all OpenUI5 libraries that the component depends on. It's crucial to list every library within the component here for correct dependency management.<br>
    In our component we are currenetly only using the two libraries `sap.ui.core` and `sap.m`. The `sap.ui.core` library provides the basic framework functionality and is required for any OpenUI5 application.<br>
    By default, loading of libraries is set to `"lazy": false`, which means they are loaded immediately when the component initializes. For libraries that are essential for your app to function from the start, like `sap.ui.core` and in our case also `sap.m` (since it is used in the root view), we can keep the default setting of `"lazy": false`.

    > 📌 **Important:** <br>

>     > It is crucial to be mindful of the lazy loading configuration for libraries. Only libraries that are absolutely necessary for your component to start should be declared with `"lazy": false`. For libraries that are not required immediately, it is recommended to override the default setting and set `"lazy": true`. This approach allows for better performance and faster initial loading of the component by deferring the loading of non-essential libraries until they are actually needed.

Reviewer 2: LGTM. Should we add a link to the [`Lib.load`](https://ui5.sap.com/#/api/sap.ui.core.Lib%23methods/sap.ui.core.Lib.load) documentation here?

maybe something like this?

> Before using a library marked with `"lazy": true`, you need to ensure that it is loaded by calling the asynchronous [`Lib.load`](https://ui5.sap.com/#/api/sap.ui.core.Lib%23methods/sap.ui.core.Lib.load) API. A usage example can be found in the linked API reference.

-   `dependencies`\(mandatory\): This section defines the dependencies of the component. It comes with the following sub-settings:

    -   The `minUI5Version` property is mandatory as it specifies the minimum version of OpenUI5 that the component requires to function properly. Following the guideline discussed in step 1, our component should be designed to work with the long-term support (LTS) version 1.120 of OpenUI5.
    
    -    The `libs` settings is used to declare all OpenUI5 libraries that the component depends on. It's crucial to list every library within the component here for correct dependency management.<br>

>     In our component we are currenetly only using the two libraries `sap.ui.core` and `sap.m`. The `sap.ui.core` library provides the basic framework functionality and is required for any OpenUI5 application.<br>

Reviewer: ```suggestion
    In our component, we currently only use the `sap.ui.core` and `sap.m` libraries. The `sap.ui.core` library provides the basic framework functionality and is required for any OpenUI5 application.<br>
```

### Conventions

> -   The manifest file is named `manifest` and stored as JSON file.

Reviewer: ```suggestion
-   The manifest file is named `manifest` and stored as a JSON file.
```
